lib/dispatcher.js

```diff
@@ -0,0 +1,19 @@
+'use strict'
+
+const EventEmitter = require('events')
+
+class Dispatcher extends EventEmitter {
+  dispatch () {
+    throw new Error('not implemented')
+  }
+
+  close () {
+    throw new Error('not implemented')
+  }
+
+  destroy () {
+    throw new Error('not implemented')
+  }
+}
```

Comment:
Why is this needed? Those are just stubs.

Reply:
not really needed

Reply:
Let's keep it, it would help TS and friends.

Reply:
Since this is only at the type definition level then I think we could even get away with something like this in the `.d.ts` files:

```ts
interface Dispatcher extends EventEmitter {
  dispatch: () => void;
  close: () => void;
  destroy: () => void;
}

class Client extends Dispatcher {}
class Agent extends Dispatcher {}
```

Reply:
but then again, I do enjoy the abstract class pattern to this. I'm +1 for keeping it since it doesn't have any perf impact